crates/sui-json-rpc-types/src/sui_object.rs

```diff
@@ -23,19 +24,37 @@ use sui_types::base_types::{
     MoveObjectType, ObjectDigest, ObjectID, ObjectInfo, ObjectRef, ObjectType, SequenceNumber,
     TransactionDigest,
 };
-use sui_types::error::{UserInputError, UserInputResult};
+use sui_types::error::{SuiObjectResponseError, UserInputError, UserInputResult};
 use sui_types::gas_coin::GasCoin;
 use sui_types::move_package::{MovePackage, TypeOrigin, UpgradeInfo};
 use sui_types::object::{Data, MoveObject, Object, ObjectFormatOptions, ObjectRead, Owner};
 
-use crate::{Page, SuiMoveStruct, SuiMoveValue};
-
 #[derive(Serialize, Deserialize, Debug, JsonSchema, Clone, PartialEq, Eq)]
-#[serde(tag = "status", content = "details", rename = "ObjectRead")]
-pub enum SuiObjectResponse {
-    Exists(SuiObjectData),
-    NotExists(ObjectID),
-    Deleted(SuiObjectRef),
+pub struct SuiObjectResponse {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub data: Option<SuiObjectData>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<SuiObjectResponseError>,
+}
```

Comment:
Should we still have ObjectStatus on the TS sdk side for this? 

```
export const SuiObjectResponse = object({
  status: ObjectStatus,
  details: union([SuiObjectData, ObjectId, SuiObjectRef]),
});

export const SuiObjectResponse = object({
  data: optional(SuiObjectData),
  error: optional(string()),
});
```
@Jordan-Mysten @666lcz ?

Reply:
I'm fine with an error string but we should either just entirely remove the status, or also encode it into the error (my preference)